src/provider.c

```diff
@@ -613,9 +613,12 @@ int p11prov_ctx_cache_sessions(P11PROV_CTX *ctx)
     return ctx->cache_sessions;
 }
 
-bool p11prov_ctx_no_operation_state(P11PROV_CTX *ctx)
+bool p11prov_ctx_is_call_blocked(P11PROV_CTX *ctx, uint64_t mask)
 {
-    return ctx->no_operation_state;
+    if (ctx->blocked_calls & mask) {
+        return true;
+    }
+    return false;
```

Comment:
could be simpler like this:
```suggestion
    return (ctx->blocked_calls & mask);
```

Reply:
true, and without the parens even... will change